libtelemetry/src/full/java/com/mapbox/android/telemetry/location/LocationUpdatesBroadcastReceiver.java

```diff
@@ -39,15 +39,20 @@ public void onReceive(Context context, Intent intent) {
       }
 
       LocationCollectionClient collectionClient =  LocationCollectionClient.getInstance();
-      MapboxTelemetry telemetry = collectionClient.getTelemetry();
-      String sessionId = collectionClient.getSessionId();
+      final MapboxTelemetry telemetry = collectionClient.getTelemetry();
+      final String sessionId = collectionClient.getSessionId();
       List<Location> locations = result.getLocations();
-      for (Location location : locations) {
+      for (final Location location : locations) {
         if (isThereAnyNaN(location) || isThereAnyInfinite(location)) {
           continue;
         }
-        final String appState = AppStateUtils.getAppState(context).toString();
-        telemetry.push(LocationMapper.create(location, appState, sessionId));
+        AppStateUtils.getAppState(context, new AppStateUtils.GetAppStateCallback() {
```

Comment:
Do we want to start new thread execution for each location? Would it be better to read state first and apply for the whole location array?

Reply:
Defenitely.